nwb_conversion_tools/utils/recordingextractordatachunkiterator.py

```diff
@@ -0,0 +1,35 @@
+from typing import Tuple, Iterable
+import numpy as np
+
+from spikeextractors import RecordingExtractor
+
+from .genericdatachunkiterator import GenericDataChunkIterator
+
+class RecordingExtractorDataChunkIterator(GenericDataChunkIterator):
+
+    def __init__(self, recording_extractor: RecordingExtractor, chunk_shape: tuple = (), buffer_mb: int = 20):
+        self.recording_extractor = recording_extractor
+        super().__init__(chunk_shape=chunk_shape, buffer_mb=buffer_mb)
+
+    def _get_data(self, selection: Tuple[slice]) -> Iterable:
+        return self.recording_extractor.get_traces(
+            channel_ids=list(range(selection[1], selection[1] + self.chunk_shape[1])),
+            start_frame=selection[0],
+            end_frame=selection[0] + self.chunk_shape[0]
+        )
+
```

Comment:
```suggestion
    def _get_data(self, selection: Tuple[slice]) -> Iterable:
        channel_ids = self.recording_extractor.get_channel_ids()[slice[1]]
        frame_ids = range(self.recording_get_num_frames())[slice[0]]
        return self.recording_extractor.get_traces(
            channel_ids=channel_ids ,
            start_frame=frame_ids[0],
            end_frame=frame_ids[1]
        )
```
@CodyCBakerPhD `selection[1]` and `selection[0]` would a a slice object; I'm not sure if a list command on a slice can be called. And the channel_ids values could be arbitrary and not necessary a range(num_channels)

Reply:
For the issue of extracting data from the slice object, yes that has now been fixed.

For the issue of `channel_ids`, that's a good catch and is fixed now.

For the issue of frame indices, all the `get_traces()` methods I see utilize direct slicing of frame values starting from `0` to `num_frames`. In `spikeextractors`, I think any `RecordingExtractor` object is guaranteed to have the frame indices in this fashion - what can and should be changed in the case of mismatch is the `timestamp` associated with each frame (and used in any `frame_to_time` conversion), which is not needed for trace extraction.